### PR TITLE
Publish extension to VS Marketplace when there is a new tag release

### PR DESCRIPTION
**Pull request by danimtb**
From: https://learn.microsoft.com/en-us/visualstudio/extensibility/walkthrough-publishing-a-visual-studio-extension-via-command-line?view=vs-2022#publishmanifest-file

And from: https://github.com/marketplace/actions/vs-marketplace-publisher

The extension is only uploaded to marketplace when a new tag (release) is pushed to the repository.

TODO: Add Personal Access Token to github secrets for publication

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ name: Build conan-vs-extension
 on:
   push:
     branches: [ develop2 ]
+    tags:
+      - '*'
```
Comment (danimtb) [inline .github/workflows/main.yml]: probably we would like this to run only for some specific pattern of the tag. Maybe `v2*` or `2.*`?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -41,3 +43,11 @@ jobs:
           name: conan-vs-extension.vsix
           path: bin\Release\conan-vs-extension.vsix
           if-no-files-found: error
+
+      - name: Publish release to marketplace
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
```
Comment (danimtb) [inline .github/workflows/main.yml]: probably we would like this to run only for some specific pattern of the tag. Maybe `refs/tags/v2` or `refs/tags/2`?

**Comment by danimtb** [comment]
Added VS Marketplace Personal Access Token as github secret

**Review by AbrilRBS — APPROVED** [review]

File: vsixManifest.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "identity": {
+        "internalName": "conan-vs-extension"
+    },
+    "overview": "README.md",
+    "priceCategory": "free",
+    "publisher": "Conan",
+    "private": false,
+    "qna": false,
+    "repo": "https://github.com/conan-io/conan-vs-extension"
+}
```
Comment (czoido) [inline vsixManifest.json]:
```suggestion
}

```

**Review by czoido — APPROVED** [review]

**Pull request merged**